docs/source/reference/types.rst

```diff
@@ -0,0 +1,233 @@
+Essentially "regular" implicits only try to figure out the type without adding any extra information.
+
+Default implicits, on the other hand, instead of trying to figure out a value, will instead just use a default, which means that unnecessary "simple" functions are unnecessary in Idris: you just use the default value.
+
+The syntax for default implicits is ``{default val x : a} -> b``, where ``val : a``, which is the default value for ``x``.
+
+The third kind of implicits are auto implicits.
+Auto implicits are most useful in proofs, where they differ from regular implicits by irrelevance.
+That is, a regular default will never initiate *anything* that cannot otherwise be initiated, while an ``auto`` default will not change anything outside of the input that cannot be initiated.
+
+For instance, if we want to automatically pass a proof through some functions, we use ``auto`` defaults.
+
+.. hint::
+    Lowercase names in types are automatically universally quantified in types.
+    For instance, ``map : (a -> b) -> List a -> List b`` is transformed into ``map : {0 a : Type} -> {0 b : Type} -> (a -> b) -> List a -> List b`` [#f1]_.
+    You can write this signature explictly and it will compile the same as the simple version, and in addition you could also have the ``forall`` keyword and wrote this as ``map : forall a . forall b . ((a -> b) -> List a -> List b)``.
+
+    If desired, this behavior can be toggled off by ``%unbound_implicits off`` (and back on by ``%unbound_implicits on``)
+
+^^^^^^^^^^^^^
+Quantities
+^^^^^^^^^^^^^
+
+Idris 2 uses Quantitative Type Theory (`QTT`_) as its theoretical basis.
+In QTT, every function type has an associated multiplicity, ``0``, ``1``, or ω.
+These correspond with how many times the given input is "used" in the result type.
+
+Very roughly, ``0`` corresponds to a value "being ignored" or *erased*, a value of ``1`` corresponds to a value being "used" once, or being *one-to-one*, and ω (which, in Idris syntax is ommited) corresponds to a type having an unrestricted number of uses.
+
+Note that this idea of "usage" is not always cut and dry.
+For instance, as we saw above, ``{0 a : Type} -> {0 b : Type} -> (a -> b) -> List a -> List b`` is a valid type, despite the fact it uses an erased arguement.
+This is because of the fact that erased arguements are only ignored at runtime, however, in the above, the usages of ``a`` and ``b`` are in a type, so they are fine.
+In addition, we can alias values without changing their multiplicity, so long as we treat them as the same value.
+For instance, if ``1 x : a``, ``let 1 y = x in EXPR`` is valid so long as exactly *one of* ``x`` or ``y`` is used in ``EXPR``.
+
+The syntax for specifying multiplicity in functions is ``(# x : a) -> B`` (or any variation on such), where ``#`` is the multiplicty.
+In addition, we may also declare ``let`` binding with a multiplictity, with the syntax ``let # x : a = val in EXPR`` or ``let # x = val in EXPR``
```

Comment:
I didn't quite mean to add the mention of `let`, since there are a number of other places where you'd do this (e.g. in a "type alias")
```idris
0 Numbers : Type
Numbers = List Int
```
I more mentioned it to point out that function parameters isn't the only place this happens